wasitest/socket.go

```diff
@@ -820,8 +910,10 @@ func testSocketConnectOK(family wasi.ProtocolFamily, typ wasi.SocketType, bind w
 		addr, errno := sys.SockConnect(ctx, sock, bind)
 		assertNotEqual(t, addr, nil)
 		assertEqual(t, addr.Family(), bind.Family())
-		if errno != wasi.ESUCCESS {
+		if typ == wasi.StreamSocket {
 			assertEqual(t, errno, wasi.EINPROGRESS)
+		} else {
+			assertEqual(t, errno, wasi.ESUCCESS)
 		}
```

Comment:
Is this true? I think connecting to the loopback may succeed immediately?

Reply:
The test connects to the loopback and it seems to always give EINPROGRESS on both darwin and linux.